File: Venkatesh/Bubble-Sort/BubbleSort.py
def bubble_sort(array):
    count = 0
    for i in range(len(array)):
        for j in range(len(array)-i-1):
            if array[j] > array[j+1]:
                array[j], array[j+1] = array[j+1], array[j]
                count += 1
    return count

arr = [15, 8, 26, 4, 62, 41, 32, 9, 21, 10]
count = bubble_sort(arr)

print("Sorted array:", arr)
print("Number of swaps:", count)

File: Aditya Vardhan Singh/Challenge Solutions/Day 1/Java/BubbleSort.java
class BubbleSort {
  int bubbleSortAndCount(int[] arr, int n) {
    boolean swap_flag = false;
    int swaps = 0;
    int temp;

    for (int i = 0; i < n; i++) {
      swap_flag = false;
        for (int j = 0; j < n - i - 1; j++) {
          if (arr[j] > arr[j+1]) {
            // swap
            temp = arr[j];
            arr[j] = arr[j+1];
            arr[j+1] = temp;
            // count swaps
            swaps++;
            // trun off flag
            swap_flag = true;
          }
        }
        if (!swap_flag) break;
    }

    return swaps;
  }

  void printArray(int[] arr, int n) {
    System.out.print("[ ");
    for (int i = 0; i < n; i++) {
      System.out.printf("%d ", arr[i]);
    }
    System.out.println("]");
  }

  public static void main(String[] args) {
    int[] arr = {64, 34, 25, 12, 22, 11, 90};
    int n = 7;

    BubbleSort ob = new BubbleSort();

    ob.printArray(arr, n);
    int swaps = ob.bubbleSortAndCount(arr, n);
    System.out.print("Sorted array: ");
    ob.printArray(arr, n);
    System.out.printf("Number of swaps: %d\n", swaps);
  }
}
